lib/plugins/nwc/src/lib.rs

```diff
@@ -181,53 +218,114 @@ impl SdkNwcService {
         Ok(Keys::parse(&secret_key)?)
     }
 
-    async fn handle_event(ctx: &RuntimeContext, notification: &RelayPoolNotification) {
-        let RelayPoolNotification::Event { event, .. } = notification else {
-            return;
-        };
-        info!("Received NWC event: {event:?}");
-
+    async fn handle_event_inner(
+        ctx: &RuntimeContext,
+        active_connections: &mut HashMap<String, ActiveConnection>,
+        event: &Event,
+    ) -> NwcResult<()> {
+        // Verify client belongs in active connections list
         let client_pubkey = event.pubkey;
+        let Some((connection_name, client)) = active_connections
+            .iter_mut()
+            .find(|(_, con)| con.pubkey == client_pubkey)
+        else {
+            return Err(NwcError::PubkeyNotFound {
+                pubkey: client_pubkey.to_string(),
+            });
+        };
 
         // Verify the event has not expired
         if event
             .tags
             .expiration()
             .is_some_and(|t| *t > Timestamp::now())
         {
-            warn!("Event {} has expired. Skipping.", event.id);
-            return;
+            return Err(NwcError::EventExpired);
```

Comment:
I think the expiration comparison is wrong.  Seems like it should be:
`.expiration().is_some_and(|t| *t < Timestamp::now())`